### PR TITLE
Replace tracking env var with _EnvironmentVariable class

### PR DESCRIPTION
**Pull request by gabrielfu**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
@gabrielfu Thanks for the PR! https://github.com/mlflow/mlflow/pull/8840 will be merged soon.

**Comment by gabrielfu** [comment]
> @gabrielfu Thanks for the PR! #8840 will be merged soon and `_TRACKING_URI_ENV_VAR` will be replaced

Ah I see! Let me revert some of it

File: mlflow/environment_variables.py [hunk 1]
```diff
@@ -286,3 +286,22 @@ def get(self):
 #: Specifies the password used to authenticate with a tracking server.
 #: (default: ``None``)
 MLFLOW_TRACKING_PASSWORD = _EnvironmentVariable("MLFLOW_TRACKING_PASSWORD", str, None)
+
+#: Specifies and takes precedence for setting the basic/bearer auth on http requests.
+MLFLOW_TRACKING_TOKEN = _EnvironmentVariable("MLFLOW_TRACKING_TOKEN", str, None)
```
Comment (harupy) [inline mlflow/environment_variables.py:291]:
```suggestion
#: Specifies and takes precedence for setting the basic/bearer auth on http requests.
#: (default: ``None``)
MLFLOW_TRACKING_TOKEN = _EnvironmentVariable("MLFLOW_TRACKING_TOKEN", str, None)
```

Can we add the default value?

File: mlflow/environment_variables.py [hunk 2]
```diff
@@ -286,3 +286,22 @@ def get(self):
...
+
+#: Specifies whether to verify TLS connection in 'requests.request' function,
+#: see https://requests.readthedocs.io/en/master/api/
+MLFLOW_TRACKING_INSECURE_TLS = _BooleanEnvironmentVariable("MLFLOW_TRACKING_INSECURE_TLS", False)
+
+#: Sets the ``verify`` param in 'requests.request' function,
```
Comment (harupy) [inline mlflow/environment_variables.py:297]:
```suggestion
#: Sets the ``verify`` param in ``requests.request`` function,
```

File: mlflow/environment_variables.py (hunk 1 above)
Comment (gabrielfu) [reply mlflow/environment_variables.py]: sure

**Review by harupy — APPROVED** [review]
LGTM!

**Comment by harupy** [comment]
@gabrielfu Merged! Feel free to replace other environment variables (you can search `= "MLFLOW_`) if you want.

**Comment by gabrielfu** [comment]
> @gabrielfu Merged! Feel free to replace other environment variables (you can search `= "MLFLOW_`) if you want.

sure! I will work on that :)

**Comment by gabrielfu** [comment]
> @gabrielfu Thanks! I'm also working on this. I'll ping you if I create PRs. I want to close #6379. This is not urgent at all, let's tackle this step by step like I do rather than filing one giant PR.

Definitely! I'll let you know too on any new changes

**Pull request merged**